voice-interface/src/main/java/com/derpgroup/derpwizard/voice/model/AlexaInput.java

```diff
@@ -20,77 +20,81 @@
 
 package com.derpgroup.derpwizard.voice.model;
 
-import java.util.HashMap;
+import java.util.Collections;
+import java.util.LinkedHashMap;
 import java.util.Map;
 import java.util.Map.Entry;
 
-
-import com.amazon.speech.slu.Intent;
 import com.amazon.speech.slu.Slot;
 import com.amazon.speech.speechlet.IntentRequest;
+import com.amazon.speech.speechlet.LaunchRequest;
+import com.amazon.speech.speechlet.SessionEndedRequest;
+import com.amazon.speech.speechlet.SpeechletRequest;
 
 class AlexaInput implements VoiceInput {
-  private IntentRequest request;
-  private Map<String,Object> metadata;
-  
+  private SpeechletRequest request;
+  private Map<String, Object> metadata = Collections.emptyMap();
+
   public AlexaInput(Object object) {
     this(object, null);
   }
 
-  @SuppressWarnings("unchecked")
-  public AlexaInput(Object request, Object metadata) {
-    if (!(request instanceof IntentRequest)) {
-      throw new IllegalArgumentException("First argument is not an instance of IntentRequest: " + request);
+  public AlexaInput(Object object, Map<String, Object> metadata) {
+    if (!(object instanceof SpeechletRequest)) {
+      throw new IllegalArgumentException("Argument is not an instance of SpeechletRequest: " + object);
     }
 
-    this.request = (IntentRequest) request;
-    
-    if(metadata != null){
-      if (!(metadata instanceof Map<?,?>)) {
-        throw new IllegalArgumentException("Second argument is not an instance of Map<?,?>: " + metadata);
-      }
-      this.metadata = (Map<String, Object>) metadata;
+    request = (SpeechletRequest) object;
+
+    if (metadata != null) {
+      this.metadata = metadata;
     }
   }
 
   @Override
-  public String getMessage() {
-    if (request.getIntent().getSlots() == null) {
-      return request.getIntent().getName();
-    }
-
-    StringBuilder buffer = new StringBuilder(request.getIntent().getName());
-    for (Entry<String, Slot> entry : request.getIntent().getSlots().entrySet()) {
-      buffer.append(' ');
-      buffer.append(entry.getValue().getValue());
+  public String getMessageAsString() {
+    if (!(request instanceof IntentRequest)) {
+      return "";
     }
 
-    return buffer.toString();
+    IntentRequest intentRequest = (IntentRequest) request;
+    return intentRequest.getIntent().getName();
```

Comment:
What I want this method to do is just return the raw text-to-speech data, but the Alexa SDK doesn't seem to support that currently.

I deleted the previous functionality because of this comment:
https://github.com/DERP-Group/derpwizard/pull/1#discussion_r42334229